File: Packet/ProtocolVersion.cs
using System.IO;

namespace Resources.Packet {
    public class ProtocolVersion : Packet {
        public int version;

        public ProtocolVersion() : base(PacketID.Version) { }
        public ProtocolVersion(BinaryReader reader) : this() {
            version = reader.ReadInt32();
        }

        protected override void WritePacketData(BinaryWriter writer) {
            writer.Write(version);
        }
    }
}

File: src/main/kotlin/packets/ProtocolVersion.kt
package me.lastexceed.cubeworldnetworking.packets

import me.lastexceed.cubeworldnetworking.utils.*

data class ProtocolVersion(
	val version: Int
) : Packet(PacketId.ProtocolVersion) {
	override suspend fun writeTo(writer: Writer) {
		writer.writeInt(version)
	}

	companion object : CwDeserializer<ProtocolVersion> {
		override suspend fun readFrom(reader: Reader) = ProtocolVersion(reader.readInt())
	}
}